rewrite-java/src/main/java/org/openrewrite/java/internal/template/AnnotationTemplateGenerator.java

```diff
@@ -230,7 +230,11 @@ private void classDeclaration(StringBuilder before, StringBuilder after, J.Class
                 }
             }
         }
-        c = c.withBody(J.Block.createEmptyBlock()).withLeadingAnnotations(null).withPrefix(Space.EMPTY);
+        c = c.withBody(J.Block.createEmptyBlock())
+                .withLeadingAnnotations(null)
+                .withPrefix(Space.EMPTY)
+                .withKind(J.ClassDeclaration.Kind.Type.Class)
+                .withPrimaryConstructor(null);
```

Comment:
Previously we were producing invalid templates; by treating all as regular classes with an instance field we solve issues for records and enums.